docs/guides/latest/workloads/daemonset.md

```diff
@@ -0,0 +1,759 @@
+
+Below is the YAML of the Deployment that we are going to create,
+
+```console
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: demo-pvc
+  namespace: demo
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: standard
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    app: stash-demo
+  name: stash-recovered
+  namespace: demo
+spec:
+  selector:
+    matchLabels:
+      app: stash-demo
+  template:
+    metadata:
+      labels:
+        app: stash-demo
+      name: busybox
+    spec:
+      containers:
+      - args:
+        - sleep
+        - "3600"
+        image: busybox
+        imagePullPolicy: IfNotPresent
+        name: busybox
+        volumeMounts:
+        - mountPath: /source/data
+          name: source-data
+      restartPolicy: Always
+      volumes:
+      - name: source-data
+        persistentVolumeClaim:
+          claimName: demo-pvc
+```
+
+Let's create the Daemonset we have shown above.
+
+```console
+$ kubectl apply -f ./docs/examples/workloads/daemonset/recovered_daemon.yaml
+persistentvolumeclaim/demo-pvc created
+daemonset.apps/stash-demo configured
+```
+
+Now, wait for Daemonset’s pod to go into the `Running` state.
+
+```console
+$ kubectl get pod -n demo
+NAME                    READY   STATUS    RESTARTS   AGE
+stash-recovered-mnv8b   1/1     Running   0          56s
+
+```
+
+**Create RestoreSession:**
+
+Now, we need to create a `RestoreSession` crd targeting the `stash-recovered` Daemonset to restore the backed up data inside it.
+
+Below is the YAML of the `RestoreSesion` crd that we are going to create,
+
+```console
+apiVersion: stash.appscode.com/v1beta1
+kind: RestoreSession
+metadata:
+  name: dmn-restore
+  namespace: demo
+spec:
+  repository:
+    name: gcs-repo
+  rules:
+  - paths:
+    - /source/data
+  target: # target indicates where the recovered data will be stored
+    ref:
+      apiVersion: apps/v1
+      kind: DaemonSet
+      name: stash-recovered
+    volumeMounts:
+    - name:  source-data
+      mountPath:  /source/data
+```
+
+Here,
+
+`spec.repository.name` specifies the `Repository` crd that holds the backend information where our backed up data has been stored.
+
+`spec.target.ref` refers to the target workload where the recovered data will be stored.
+
+Let's create the `RestoreSession` crd we have shown above,
+
+```console
+kubectl apply -f ./docs/examples/workloads/daemonset/restoresession.yaml
+restoresession.stash.appscode.com/dmn-restore created
+```
+
+Once, you have created the `RestoreSession` crd, Stash will inject `init-container` to `stash-recovered` Daemonset. The Daemonset will restart and the `init-container` will recovered on start-up.
+
+**Verify Init-Container:**
+
+Wait until the `init-container` has been injected to the `stash-recovered` Daemonset's pod, Run the following command to describe the `stash-recovered` Daemonset's pod,
```

Comment:
DaemonSet's pod.

Reply:
Let's describe the DaemonSet to verify that `init-container` has been injected successfully.